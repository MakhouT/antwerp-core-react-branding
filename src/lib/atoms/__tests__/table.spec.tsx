import * as React from "react";

import { shallow } from "enzyme";

import * as A from "../..";
import { renderAndMatchSnapshot } from "../../../__tests__/__fixtures__/testUtils";

describe("Table", () => {
    it("Renders default", () => {
        // Act
        const element = (
            <A.Table>
                <tr><td>blabla</td></tr>
            </A.Table>
        );

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("Renders with level Success", () => {
        // Act
        const level = A.Levels.Success;
        const element = (
            <A.Table level={level}>
                <tr><td>blabla</td></tr>
            </A.Table>
        );

        // Act and assert
        const tree = renderAndMatchSnapshot(element);
        expect(tree.props.className).toContain(`a-table--${level}`);
    });

    it("Renders with level Primary", () => {
        // Act
        const level = A.Levels.Primary;
        const element = (
            <A.Table level={level}>
                <tr><td>blabla</td></tr>
            </A.Table>
        );

        // Act and assert
        const tree = renderAndMatchSnapshot(element);
        expect(tree.props.className).toContain(`a-table--${level}`);
    });

    it("Renders small table", () => {
        // Act
        const element = (
            <A.Table small>
                <tr><td>blabla</td></tr>
            </A.Table>
        );

        // Act and assert
        const tree = renderAndMatchSnapshot(element);
        expect(tree.props.className).toContain(`a-table--small`);
    });

    it("Renders striped table", () => {
        // Act
        const element = (
            <A.Table striped>
                <tr><td>blabla</td></tr>
            </A.Table>
        );

        // Act and assert
        const tree = renderAndMatchSnapshot(element);
        expect(tree.props.className).toContain(`a-table--striped`);
    });

    it("Renders children", () => {
        // Act
        const element = (
            <A.Table striped>
                <tr><td>blabla</td></tr>
                <tr><td>blabla</td></tr>
                <tr><td>blabla</td></tr>
                <tr><td>blabla</td></tr>
            </A.Table>
        );

        // Act
        const table = shallow(element);

        // Act and assert
        expect(table.find("td").length).toBe(4);
    });
});
