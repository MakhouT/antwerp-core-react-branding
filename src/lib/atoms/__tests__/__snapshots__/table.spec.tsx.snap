// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Renders default 1`] = `
<table
  className="a-table"
>
  <tr>
    <td>
      blabla
    </td>
  </tr>
</table>
`;

exports[`Table Renders small table 1`] = `
<table
  className="a-table a-table--small"
>
  <tr>
    <td>
      blabla
    </td>
  </tr>
</table>
`;

exports[`Table Renders striped table 1`] = `
<table
  className="a-table a-table--striped"
>
  <tr>
    <td>
      blabla
    </td>
  </tr>
</table>
`;

exports[`Table Renders with level Primary 1`] = `
<table
  className="a-table a-table--primary"
>
  <tr>
    <td>
      blabla
    </td>
  </tr>
</table>
`;

exports[`Table Renders with level Success 1`] = `
<table
  className="a-table a-table--success"
>
  <tr>
    <td>
      blabla
    </td>
  </tr>
</table>
`;
